modular_skyrat/modules/gunsgalore/code/guns/ballistic_master.dm

```diff
@@ -53,13 +53,62 @@
 				inhand_icon_state = "[initial(icon_state)]"
 				worn_icon_state = "[initial(icon_state)]"
 
-/obj/item/gun/ballistic/insert_magazine(mob/user, obj/item/ammo_box/magazine/AM, display_message)
-	if(reload_time && !HAS_TRAIT(user, TRAIT_INSTANT_RELOAD) && magazine) //This only happens when you're attempting a tactical reload, e.g. there's a mag already inserted.
-		to_chat(user, span_notice("You start to insert the magazine into [src]!"))
-		if(!do_after(user, reload_time, src))
-			to_chat(user, span_danger("You fail to insert the magazine into [src]!"))
-			return
+/obj/item/gun/ballistic/attackby(obj/item/A, mob/user, params)
 	. = ..()
+	if (.)
+		return
+	if (!internal_magazine && istype(A, /obj/item/ammo_box/magazine))
+		handle_magazine(user, A)
+		return
+	if (istype(A, /obj/item/ammo_casing) || istype(A, /obj/item/ammo_box))
+		if (bolt_type == BOLT_TYPE_NO_BOLT || internal_magazine)
+			if (chambered && !chambered.loaded_projectile)
+				chambered.forceMove(drop_location())
+				chambered = null
+			var/num_loaded = magazine?.attackby(A, user, params, TRUE)
+			if (num_loaded)
+				to_chat(user, span_notice("You load [num_loaded] [cartridge_wording]\s into [src]."))
+				playsound(src, load_sound, load_sound_volume, load_sound_vary)
+				if (chambered == null && bolt_type == BOLT_TYPE_NO_BOLT)
+					chamber_round()
+				A.update_appearance()
+				update_appearance()
+			return
+	if(istype(A, /obj/item/suppressor))
+		var/obj/item/suppressor/S = A
+		if(!can_suppress)
+			to_chat(user, span_warning("You can't seem to figure out how to fit [S] on [src]!"))
+			return
+		if(!user.is_holding(src))
+			to_chat(user, span_warning("You need be holding [src] to fit [S] to it!"))
+			return
+		if(suppressed)
+			to_chat(user, span_warning("[src] already has a suppressor!"))
+			return
+		if(user.transferItemToLoc(A, src))
+			to_chat(user, span_notice("You screw [S] onto [src]."))
+			install_suppressor(A)
+			return
+	if (can_be_sawn_off)
+		if (sawoff(user, A))
+			return
+
+	if(can_misfire && istype(A, /obj/item/stack/sheet/cloth))
+		if(guncleaning(user, A))
+			return
+
+	return FALSE
+
+/obj/item/gun/ballistic/proc/handle_magazine(mob/user, obj/item/ammo_box/magazine/inserting_magazine)
+	if(magazine) // If we already have a magazine inserted, we're going to begin tactically reloading it.
+		if(reload_time && !HAS_TRAIT(user, TRAIT_INSTANT_RELOAD)) // Check if we have a reload time to tactical reloading, or if we have the instant reload trait.
```

Comment:
have you considered just removing your reload time stuff instead to accomplish the same effect while removing non-modular code edits

Reply:
No, it's something we'd like to keep.